components/ai_chat/resources/page/state/data-context-provider.tsx

```diff
@@ -163,17 +162,10 @@ function DataContextProvider (props: DataContextProviderProps) {
     getPageHandlerInstance().pageHandler.getShouldSendPageContents().then(({ shouldSend }) => setShouldSendPageContents(shouldSend))
   }
 
-  const shouldShowLongPageWarning = React.useMemo(() => {
-    if (
-      !hasDismissedLongPageWarning &&
-      conversationHistory.length >= 1 &&
-      siteInfo?.isContentTruncated
-    ) {
-      return true
-    }
-
-    return false
-  }, [conversationHistory, hasDismissedLongPageWarning, siteInfo?.isContentTruncated])
+  const shouldShowLongPageWarning = React.useMemo(() =>
+    conversationHistory.length >= 1 &&
+    siteInfo?.contentUsedPercentage < 100,
+  [conversationHistory.length, siteInfo?.contentUsedPercentage])
```

Comment:
nit: this doesn't need to use `useMemo` because it's just checking the same properties as the dependencies, which will get checked in `useMemo` itself! It's fine either way...